### PR TITLE
Feature/refresh for rest api changes

### PR DESCRIPTION
**Pull request by hmmmmniek**
- Fixed mistakes in the interfaces used to send request (remove parameters that dont do anything). And added new options which were added to the REST api but not yet added to targomo-js.
- Also refactored all the traveloptions interfaces so that each service actually only exposes options which can be used and have an effect on the result.
- Added detailed comments for the whole Fleetplanner interface.
- I removed the way too detailed tests for the Fleetplanner, they did uncover some problems with the service that had to be solved, but it is not a good test to run every time you build the library.
- Finally, I split up the Multigraph into multigraph and statefulMultigraph for clarity.

### CODE REVIEW

**Comment by silberjan** [comment]
@hmmmmniek can you also as a mainenance task in this pr go through everything and remove unnecessary `console.log` i found one to be super annoying in the routes request. Ill do a full review later today

**Comment by hmmmmniek** [comment]
@silberjan I noticed that console.log when I was working on it and I had already removed it ;) https://github.com/targomo/targomo-js/pull/22/files#diff-71ec3f49dfbc16b07b497da448a55ac3
I didn't find any other console.logs. So it's clean now.

File: src/types/options/statisticsRequestOptions.ts [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+import { BaseRequestOptions } from '../requestOptions';
 
-export interface StatisticsTravelRequestOptions extends TravelRequestOptions {
-  // statistics: (StatisticsKey | number)[]
+export interface StatisticsTravelRequestOptions extends BaseRequestOptions {
   statisticsGroup: StatisticsGroupId
   inactiveSources?: LatLngId[]
   closestSources?: boolean
-  intersectionMode?: string
+  travelType: 'walk' | 'bike' | 'car'
```
Comment (silberjan) [inline src/types/options/statisticsRequestOptions.ts]: transit missing!

Comment (hmmmmniek) [reply src/types/options/statisticsRequestOptions.ts]: This is on purpose, according to https://dev.route360.net/docs-public/statistics/#/Simulation_Services/post_simulation_start_ , this statistics function doesnt support transit mode.

Comment (silberjan) [reply src/types/options/statisticsRequestOptions.ts]: i talked to @DavidRedlich, the docs are incrorrect at this point. transit is supported if it is supported in the used endpoint.

Comment (gerbsen) [reply src/types/options/statisticsRequestOptions.ts]: Can you please fix this in the docs @hmmmmniek ?

Comment (hmmmmniek) [reply src/types/options/statisticsRequestOptions.ts]: I'll change this in targomo-js here on tuesday after my vacation. However, I have never done any work on the rest api docs, not sure who is responsible for that. @gerbsen

**Pull request merged**